lib/s3_meta_sync/syncer.rb

```diff
@@ -83,6 +87,21 @@ def download(source, destination)
       end
     end
 
+    # Sometimes SIGTERM causes Dir.mktmpdir to not properly clear the temp folder
+    # Remove day old folders
+    def clear_old_temp_folders
+      path = File.join(Dir.tmpdir, STAGING_AREA_PREFIX + '*')
+
+      day = 24 * 60 * 60
+      removed = Dir.glob(path).select { |dir| older_than(dir, day) }. map { |dir| FileUtils.rm_rf(dir) }
+
+      log "Removed #{removed} old temp folder(s)" if removed.count > 0
+    end
+
+    def older_than(dir, seconds)
```

Comment:
That looks better. `older_than?` I was keeping a method so I can mock it, but will try mocking File.ctime instead